.github/workflows/build.yml

```diff
@@ -42,7 +44,58 @@ jobs:
         run: ./gradlew :harmony-kotlin:ktlintCheck :harmony-kotlin:detekt
 
       - name: Build
-        run: ./gradlew :harmony-kotlin:build --stacktrace
+        run: ./gradlew :harmony-kotlin:assemble
+
+      - name: Test
+        run: ./gradlew :harmony-kotlin:check
+
+  build-ios:
+    name: Build & test iOS
+
+    runs-on: macos-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-java@v2
+        with:
+          distribution: "adopt"
+          java-version: "11"
+
+      - name: Validate Gradle Wrapper
+        uses: gradle/wrapper-validation-action@v1
+
+      - name: Cache gradle
+        uses: actions/cache@v3
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('*.gradle') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Cache konan
+        uses: actions/cache@v3
+        with:
+          path: ~/.konan
+          key: ${{ runner.os }}-gradle-${{ hashFiles('*.gradle') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Build
+        run: >
+          ./gradlew 
+          :harmony-kotlin:compileKotlinIosArm64
+          :harmony-kotlin:compileKotlinIosSimulatorArm64
+          :harmony-kotlin:compileKotlinIosX64
+          :harmony-kotlin:iosArm64MetadataJar
+          :harmony-kotlin:iosSimulatorArm64MetadataJar
+          :harmony-kotlin:iosX64MetadataJar
```

Comment:
If you just invoke the tests task I am pretty sure that those ones under "Build" are also executed too. Any advantage on declaring them independently?

Reply:
the idea is the same than in the previous comment. I prefer to split the job in small steps.